fms/triton/moe_kernel.py

```diff
@@ -0,0 +1,422 @@
+import math
+import pathlib
+from typing import Tuple
+
+import torch
+import triton  # type: ignore[import-untyped]
+import triton.language as tl  # type: ignore[import-untyped]
+
+
+"""Fused MoE kernel."""
```

Comment:
I think we'll want unit tests for some/most of these kernels, even though they won't run automatically on github ci

Reply:
yes, I'll ask @AdnanHoque for the ones he wrote for the blog post testing this kernel. I also plan to add new TP unit tests, given that now we have a single function to test, `sharded_copy`, instead of a multitude.